### PR TITLE
avm2: Add More BitMapData methods

### LINKED ISSUE #8133: Error on https://www.newgrounds.com/portal/view/556643. Console is screaming.
### Describe the bug

I'm using the latest nightly build of Ruffle for Firefox. This happened during an attempt to play the seminal flash game [K.O.L.M. ](https://www.newgrounds.com/portal/view/556643)Upon reaching the main menu scene I get this in the console:
```
20:54:52.023 AVM2 error: Object(ErrorObject(ErrorObject(GcCell(Gc { ptr: 0x537a46d4 }))))
20:54:52.024 ReferenceError: Error #1069: Property setPixels not found on flash.display.BitmapData and there is no default value.
20:54:52.025 AVM2 error: Object(ErrorObject(ErrorObject(GcCell(Gc { ptr: 0x537a46d4 }))))
20:54:52.026 ReferenceError: Error #1069: Property setPixels not found on flash.display.BitmapData and there is no default value.
20:54:52.027 AVM2 error: Object(ErrorObject(ErrorObject(GcCell(Gc { ptr: 0x537a46d4 }))))
20:54:52.027 ReferenceError: Error #1069: Property setPixels not found on flash.display.BitmapData and there is no default value.
20:54:52.028 Error dispatching event EventObject(EventObject { type: "enterFrame", class: flash.events::Event, ptr: 0x12c0478 }) to handler FunctionObject(FunctionObject(GcCell(Gc { ptr: 0x1d6436c }))) : AvmError(Object(ErrorObject(ErrorObject(GcCell(Gc { ptr: 0x537a46d4 }))))) 
```

Thereafter, the console is reduced to outputting a series of A's and unknown characters. On top of that, the game runs so slow it is unplayable.

### Expected behavior

I would expect the main menu to reveal itself in a decent amount of time, and not at an absurdly low framerate. That, and for the console to not scream at me.

### Affected platform

Browser's extension

### Operating system

Windows 10

### Browser

Mozilla Firefox 104

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by hatal175**
This also fixes a small bug in copyChannels - While setting the color channels it didn't transform the color to the appropiate transperancy.

I also changed the calculation signs in copychannel since it panicked on large numbers. Not sure about this though. It seems like the functions can't really decide on the sign of the input.

This fixes #8133.

### CODE REVIEW

**Comment by hatal175** [comment]
Can anyone help with this error? I don't know what that's about

**Comment by Aaron1011** [comment]
@hatal175: This is unrelated to your PR - we're working on a fix.

File: core/src/avm2/globals/flash/display/bitmapdata.rs [hunk 1]
```diff
@@ -375,6 +375,330 @@ pub fn get_pixel32<'gc>(
...
+            .ok_or("ArgumentError: Parameter must be a bytearray")?;
+
+        let mut bitmap_data = bitmap_data.write(activation.context.gc_context);
+        let mut ind = 0;
+
+        for y in y..y + height {
+            for x in x..x + width {
+                // Copy data from bytearray until EOFError or finished
+                if let Ok(color) = ba_read.read_int_at(ind) {
+                    bitmap_data.set_pixel32_raw(x, y, color.into());
```
Comment (Aaron1011) [inline core/src/avm2/globals/flash/display/bitmapdata.rs:472]: I think this should be `set_pixel32, which will correctly discard the transparency information if `bitmap_data` is not opaque. Can you add a test for this?

Comment (hatal175) [reply core/src/avm2/globals/flash/display/bitmapdata.rs]: Fixed. It also made me notice I should have advanced index by 4.

**Review by kmeisthax — APPROVED** [review]

**Pull request merged**